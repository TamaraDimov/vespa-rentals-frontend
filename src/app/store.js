import { configureStore } from '@reduxjs/toolkit';
import userSlice from '../redux/reducers/userSlice';
import motorcycleReducer from '../redux/reducers/motorcycleSlice';
import reservationReducer from '../redux/reducers/reservationSlice';

const store = configureStore({
  reducer: {
    motorcycle: motorcycleReducer,
    reservation: reservationReducer,
    user: userSlice,
  },
});

export default store;
