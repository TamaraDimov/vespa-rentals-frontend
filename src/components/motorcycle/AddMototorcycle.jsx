import { useState } from 'react';
import { MdAddPhotoAlternate, MdSportsMotorsports } from 'react-icons/md';
import { FaMotorcycle } from 'react-icons/fa6';
import { useDispatch, useSelector } from 'react-redux';
import { adddMotorcycle } from '../../redux/reducers/motorcycleSlice';
import Alerts from './subcomponents/pop/Alert';
import Created from './subcomponents/pop/Created';
import './AddMotorcycle.css';

export default function AddMotorcycle() {
  const [name, setName] = useState('');
  const [model, setModel] = useState('');
  const [photo, setPhoto] = useState('');
  const [description, setDescription] = useState('');
  const dispatch = useDispatch('');
  const { returnedMessage } = useSelector((state) => state.motorcycle);

  const handleSubmission = (e) => {
    e.preventDefault();
    dispatch(
      adddMotorcycle({
        name,
        model,
        photo,
        description,
      }),
    );
  };

  return (
    <div className="add-motorcycle-bg">
      <div className="add-motorcycle-container">
        <h3 className="form-title2">Add New Motorcycle</h3>
        <form className="form-2" onSubmit={handleSubmission}>
          <div
            className="wrap-input1002 validate-input2"
            data-validate="Enter username"
          >
            <input
              type="text"
              placeholder="Photo"
              name="photo"
              onChange={(e) => setPhoto(e.target.value)}
              value={photo}
              className="input1002"
            />

            <span className="focus-input1002" data-placeholder="&#xf207;">
              <MdAddPhotoAlternate className="input-icon2" />
            </span>
          </div>
          <div
            className="wrap-input1002 validate-input2"
            data-validate="Enter email"
          >
            <input
              name="name"
              placeholder="Name"
              type="text"
              onChange={(e) => setName(e.target.value)}
              value={name}
              className="input1002"
            />
            <span className="focus-input1002" data-placeholder="&#xf207;">
              <MdSportsMotorsports className="input-icon2" />
            </span>
          </div>

          {/* <div
            className="wrap-input100 validate-input"
            data-validate="Enter password"
          >
            <span className="focus-input100" data-placeholder="&#xf207;">
              <RiLockPasswordFill className="input-icon" />
            </span>
          </div> */}

          <div
            className="wrap-input1002 validate-input2"
            data-validate="Enter email"
          >
            <input
              type="text"
              placeholder="Model"
              name="model"
              onChange={(e) => setModel(e.target.value)}
              value={model}
              className="input1002"
            />
            <span className="focus-input1002" data-placeholder="&#xf207;">
              <FaMotorcycle className="input-icon2" />
            </span>
          </div>

          <textarea
            placeholder="Description"
            onChange={(e) => setDescription(e.target.value)}
            value={description}
            className="custom-textarea"
          />

          <div className="d3button-container">
            <span className="span-3dbutton">
              <button type="submit" className="motor-form-btn">
                {/* {returnedMessage === 'pending' ? 'Sending...' : 'Submit'} */}
              </button>
            </span>
            {returnedMessage === 'created' && <Created />}
            {returnedMessage === 'Unprocessable Entity' && <Alerts />}
          </div>
        </form>
      </div>
    </div>
  );
}
