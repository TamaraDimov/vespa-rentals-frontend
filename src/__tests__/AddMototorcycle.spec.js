import { create } from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../app/store';
import AddMototorcycle from '../components/motorcycle/AddMototorcycle';

describe('AddMototorcycle component', () => {
  test('renders correctly', () => {
    const component = create(
      <Provider store={store}>
        <Router>
          <AddMototorcycle />
        </Router>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
