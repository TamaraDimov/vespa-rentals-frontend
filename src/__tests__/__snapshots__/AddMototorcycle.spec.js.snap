// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMototorcycle component renders correctly 1`] = `
<div
  className="add-motorcycle-bg"
>
  <div
    className="add-motorcycle-container"
  >
    <h3
      className="form-title2"
    >
      Add New Motorcycle
    </h3>
    <form
      className="form-2"
      onSubmit={[Function]}
    >
      <div
        className="wrap-input1002 validate-input2"
        data-validate="Enter username"
      >
        <input
          className="input1002"
          name="photo"
          onChange={[Function]}
          placeholder="Photo"
          type="text"
          value=""
        />
        <span
          className="focus-input1002"
          data-placeholder=""
        >
          <svg
            className="input-icon2"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M19 7v2.99s-1.99.01-2 0V7h-3s.01-1.99 0-2h3V2h2v3h3v2h-3zm-3 4V8h-3V5H5c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2v-8h-3zM5 19l3-4 2 3 3-4 4 5H5z"
            />
          </svg>
        </span>
      </div>
      <div
        className="wrap-input1002 validate-input2"
        data-validate="Enter email"
      >
        <input
          className="input1002"
          name="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value=""
        />
        <span
          className="focus-input1002"
          data-placeholder=""
        >
          <svg
            className="input-icon2"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 11.39c0-.65-.39-1.23-.98-1.48L5.44 7.55c-1.48 1.68-2.32 3.7-2.8 5.45h7.75c.89 0 1.61-.72 1.61-1.61z"
            />
            <path
              d="M21.96 11.22c-.41-4.41-4.56-7.49-8.98-7.2-2.51.16-4.44.94-5.93 2.04l4.74 2.01c1.33.57 2.2 1.87 2.2 3.32 0 1.99-1.62 3.61-3.61 3.61H2.21C2 16.31 2 17.2 2 17.2v.8c0 1.1.9 2 2 2h10c4.67 0 8.41-4.01 7.96-8.78z"
            />
          </svg>
        </span>
      </div>
      <div
        className="wrap-input1002 validate-input2"
        data-validate="Enter email"
      >
        <input
          className="input1002"
          name="model"
          onChange={[Function]}
          placeholder="Model"
          type="text"
          value=""
        />
        <span
          className="focus-input1002"
          data-placeholder=""
        >
          <svg
            className="input-icon2"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 640 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280 32c-13.3 0-24 10.7-24 24s10.7 24 24 24h57.7l16.4 30.3L256 192l-45.3-45.3c-12-12-28.3-18.7-45.3-18.7H64c-17.7 0-32 14.3-32 32v32h96c88.4 0 160 71.6 160 160c0 11-1.1 21.7-3.2 32h70.4c-2.1-10.3-3.2-21-3.2-32c0-52.2 25-98.6 63.7-127.8l15.4 28.6C402.4 276.3 384 312 384 352c0 70.7 57.3 128 128 128s128-57.3 128-128s-57.3-128-128-128c-13.5 0-26.5 2.1-38.7 6L418.2 128H480c17.7 0 32-14.3 32-32V64c0-17.7-14.3-32-32-32H459.6c-7.5 0-14.7 2.6-20.5 7.4L391.7 78.9l-14-26c-7-12.9-20.5-21-35.2-21H280zM462.7 311.2l28.2 52.2c6.3 11.7 20.9 16 32.5 9.7s16-20.9 9.7-32.5l-28.2-52.2c2.3-.3 4.7-.4 7.1-.4c35.3 0 64 28.7 64 64s-28.7 64-64 64s-64-28.7-64-64c0-15.5 5.5-29.7 14.7-40.8zM187.3 376c-9.5 23.5-32.5 40-59.3 40c-35.3 0-64-28.7-64-64s28.7-64 64-64c26.9 0 49.9 16.5 59.3 40h66.4C242.5 268.8 190.5 224 128 224C57.3 224 0 281.3 0 352s57.3 128 128 128c62.5 0 114.5-44.8 125.8-104H187.3zM128 384a32 32 0 1 0 0-64 32 32 0 1 0 0 64z"
            />
          </svg>
        </span>
      </div>
      <textarea
        className="custom-textarea"
        onChange={[Function]}
        placeholder="Description"
        value=""
      />
      <div
        className="d3button-container"
      >
        <span
          className="span-3dbutton"
        >
          <button
            className="motor-form-btn"
            type="submit"
          />
        </span>
      </div>
    </form>
  </div>
</div>
`;
